packages/terra-framework-docs/src/terra-dev-site/test/table/TableWithCollapsibleSectionsAndSubSections.test.jsx

```diff
@@ -0,0 +1,172 @@
+          {
+            id: '3',
+            cells: [
+              { content: 'Fleck, Arthur' },
+              { content: '1007-MTN' },
+              { content: 'Unstable' },
+              { content: 'Inpatient, 2 months' },
+              { content: '' },
+              { content: 'Quinzell, Harleen' },
+            ],
+          },
+          {
+            id: '4',
+            cells: [
+              { content: 'Wayne, Bruce' },
+              { content: '1007-MTN-DR' },
+              { content: 'Stable' },
+              { content: 'Outpatient, 2 days' },
+              { content: 'Phytochemicals' },
+              { content: 'Grayson, Richard' },
+            ],
+          },
+        ],
+      },
+    ],
+  },
+  {
+    id: 'section-1',
+    isCollapsible: true,
+    isCollapsed: true,
+    text: 'Test Section #2',
+    subsections: [
+      {
+        id: 'subsection-2',
+        text: 'Test SubSection',
+        rows: [
+          {
+            id: '5',
+            cells: [
+              { content: 'Parker, Peter' },
+              { content: '1007-MTN' },
+              { content: 'Unstable' },
+              { content: 'Inpatient, 2 months' },
+              { content: '' },
+              { content: 'Octopus, Doctor' },
+            ],
+          },
+          {
+            id: '6',
+            cells: [
+              { content: 'Stark, Tony' },
+              { content: '1007-MTN-DR' },
+              { content: 'Stable' },
+              { content: 'Outpatient, 2 days' },
+              { content: 'Phytochemicals' },
+              { content: 'America, Captain' },
+            ],
+          },
+        ],
+      },
+      {
+        id: 'subsection-3',
+        text: 'Test SubSection #2',
+        rows: [
+          {
+            id: '7',
+            cells: [
+              { content: 'Parker, Peter' },
+              { content: '1007-MTN' },
+              { content: 'Unstable' },
+              { content: 'Inpatient, 2 months' },
+              { content: '' },
+              { content: 'Octopus, Doctor' },
+            ],
+          },
+          {
+            id: '8',
+            cells: [
+              { content: 'Stark, Tony' },
+              { content: '1007-MTN-DR' },
+              { content: 'Stable' },
+              { content: 'Outpatient, 2 days' },
+              { content: 'Phytochemicals' },
+              { content: 'America, Captain' },
+            ],
+          },
+        ],
+      },
+    ],
+  }],
+};
+
+const TableWithCollapsibleSections = () => {
+  const [tableSections, setTableSections] = useState(tableData.sections);
+
+  const handleSectionSelect = (sectionId) => {
+    const newSections = [...tableSections];
+    const sectionIndex = newSections.findIndex(section => section.id === sectionId);
+
+    if (sectionIndex > -1) {
+      newSections[sectionIndex].isCollapsed = !newSections[sectionIndex].isCollapsed;
+    }
+
+    setTableSections(newSections);
+  };
+
+  return (
+    <Table
```

Comment:
I would add this example to user facing part of the documentation as well, so that users have access to the implementation code example.